website/client-old/js/controllers/inventoryCtrl.js

```diff
@@ -382,6 +382,24 @@ habitrpg.controller("InventoryCtrl",
       });
     };
 
+    $scope.shouldShowPremiumMountRow = function (potion) {
+      potion = Content.premiumHatchingPotions[potion].key;
+      var pets = Object.keys(user.items.pets);
+      return pets.find(function (pet) {
+        return pet.indexOf(potion) !== -1;
+      });
+    };
+
+    $scope.shouldShowPremiumMountSection = function () {
```

Comment:
Made the changes, including on the `shouldShowPremiumPetRow` function. Hope I did the way you wanted!
On a side note, what did you mean by "It's all function as is"?